trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/ConfigurationManagerService.java

```diff
@@ -155,11 +152,17 @@ public Response postConfiguration(@Parameter(description = CFG_TYPE_ID) @PathPar
             return Response.status(Status.BAD_REQUEST).entity(EndpointConstants.MISSING_PARAMETERS).build();
         }
 
-        @SuppressWarnings("null")
-        @NonNull Map<@NonNull String, @NonNull Object> params = queryParameters.getParameters();
-
+//        @NonNull Map<@NonNull String, @NonNull Object> params = new HashMap<>();
+//        if (configurationSource.getConfigurationSourceType().getSchemaFile() != null) {
+//            // Pass JSON object as json string
+//            params.put(TmfConfiguration.JSON_STRING_KEY, queryParameters.getParameters().toString());
+//        } else {
+//            // Convert it to a Map<String, Object>
+//            ObjectMapper mapper = new ObjectMapper();
+//            params = mapper.convertValue(queryParameters.getParameters(), new TypeReference<Map<String,Object>>(){});
+//        }
         try {
-            ITmfConfiguration config = configurationSource.create(params);
+            ITmfConfiguration config = configurationSource.create(queryParameters.getParameters().toString());
```

Comment:
API will be changed to use Map<String, Object> instead of json string (see discussion for reasoning: https://github.com/eclipse-tracecompass-incubator/org.eclipse.tracecompass.incubator/pull/88#discussion_r1791829115)